SU2_CFD/src/numerics/NEMO/CNEMONumerics.cpp

```diff
@@ -330,299 +334,72 @@ void CNEMONumerics::GetViscousProjFlux(su2double *val_primvar,
+
+    case 6:
+      switch (nSpecies) {
+        case 1: return ComputeViscousJacs_impl<6, 1>(val_Mean_PrimVar,
+                                                     val_Mean_Eve, val_Mean_Cvve, val_diffusion_coeff,
+                                                     val_laminar_viscosity, val_eddy_viscosity,
+                                                     val_thermal_conductivity, val_thermal_conductivity_ve,
+                                                     val_dist_ij, val_normal, val_dS, val_Fv,
+                                                     val_Jac_i, val_Jac_j, config);
+
+        case 2: return ComputeViscousJacs_impl<6, 2>(val_Mean_PrimVar,
+                                                     val_Mean_Eve, val_Mean_Cvve, val_diffusion_coeff,
+                                                     val_laminar_viscosity, val_eddy_viscosity,
+                                                     val_thermal_conductivity, val_thermal_conductivity_ve,
+                                                     val_dist_ij, val_normal, val_dS, val_Fv,
+                                                     val_Jac_i, val_Jac_j, config);
+
+        default: SU2_MPI::Error("nVar and nSpecies mismatch.", CURRENT_FUNCTION);
+      }
+    break;
+    case 7:
+      return ComputeViscousJacs_impl<7, 2>(val_Mean_PrimVar,
+                                           val_Mean_Eve, val_Mean_Cvve, val_diffusion_coeff,
+                                           val_laminar_viscosity, val_eddy_viscosity,
+                                           val_thermal_conductivity, val_thermal_conductivity_ve,
+                                           val_dist_ij, val_normal, val_dS, val_Fv,
+                                           val_Jac_i, val_Jac_j, config);
+
+     case 9:
+      return ComputeViscousJacs_impl<9, 5>(val_Mean_PrimVar,
+                                           val_Mean_Eve, val_Mean_Cvve, val_diffusion_coeff,
+                                           val_laminar_viscosity, val_eddy_viscosity,
+                                           val_thermal_conductivity, val_thermal_conductivity_ve,
+                                           val_dist_ij, val_normal, val_dS, val_Fv,
+                                           val_Jac_i, val_Jac_j, config);
+
+    case 10:
+      return ComputeViscousJacs_impl<10, 5>(val_Mean_PrimVar,
+                                            val_Mean_Eve, val_Mean_Cvve, val_diffusion_coeff,
+                                            val_laminar_viscosity, val_eddy_viscosity,
+                                            val_thermal_conductivity, val_thermal_conductivity_ve,
+                                            val_dist_ij, val_normal, val_dS, val_Fv,
+                                            val_Jac_i, val_Jac_j, config);
+
+    default:
+      SU2_MPI::Error("Implicit solver not setup for seleced Gas Model.", CURRENT_FUNCTION);
```

Comment:
Yes, oops.  

Now this brings me to the question of why have many options with a fallback of DynamicSize instead of 1/none options with the other being the dynamic size